src/domain/Document/Document.ts

```diff
@@ -73,4 +74,22 @@ export class Document extends BaseEntity {
       this.numberOfIdentifiedArticles = props.numberOfIdentifiedArticles;
     if (props.attachments) this.attachments = props.attachments;
   }
+
+  /**
+   * Assign responsible on a document.
+   * The assigner is required to be at least a LSE or LSS as per specification.
+   * This is verified at the controller level.
+   *
+   * The asignee is a LSS or LSE as per current specification.
+   * This is a domain constraint enforced here.
+   *
+   * @param user
+   */
+  assignResponsible(user: User): boolean {
+    if (user.role !== Role.LSS && user.role !== Role.LSE)
+      throw new Error('user to be assigned does not have the required LSS or LSE role');
+
+    this.assignedUser = user.id;
+    return true;
```

Comment:
I don't believe this method should return anything. Anywhere this method is used we have to wrap it in a try catch due to the role validation error that might be thrown thus we don't ever care to perform an addition check based on the result of this method.